tests/diagnostic/test_diagnostics.h

```diff
@@ -35,9 +36,73 @@ std::array<std::uint32_t, GridLayout::dimension> fieldIndices(FieldFilter ff, Fu
         return {get(core::Direction::X), get(core::Direction::Y), get(core::Direction::Z)};
 }
 
+
+template<std::size_t dim, typename T, typename GridLayout, typename Field>
+void validateFluidGhosts(std::vector<T> const& data, GridLayout const& layout, Field const& field)
+{
+    using Filter = FieldDomainPlusNFilter;
+    auto filter  = Filter{1}; // include one ghost on each side
+
+    auto beg = fieldIndices(filter, std::mem_fn(&Filter::template start<Field, GridLayout>), layout,
+                            field);
+    auto end = fieldIndices(filter, std::mem_fn(&Filter::template end<Field, GridLayout>), layout,
+                            field);
+    auto siz = fieldIndices(FieldNullFilter{},
+                            std::mem_fn(&FieldNullFilter::template size<Field, GridLayout>), layout,
+                            field);
+
+    for (std::size_t d = 0; d < dim; d++)
+    {
+        ASSERT_TRUE(beg[d] > 0);
+        ASSERT_TRUE(end[d] > beg[d]);
+        ASSERT_TRUE(siz[d] > end[d]);
+    }
+
+    core::NdArrayView<dim, float> view{data, siz};
+
+    if constexpr (dim == 1)
+    {
+        std::size_t expected_nans = ((siz[0] - end[0]) * 2) - 2;
```

Comment:
unused?
test should assert first nbrGhost-1 ghost nodes are indeed nans
Also : assumes nbrGhost == 5, which may not be true in future...

Reply:
the test needs updating like it was with the hollow_box so, maybe the Mask stuff can return the total number of cells it covers

not sure where the assumption of 5 is here

Reply:
I misunderstood there is no further assumption that "we nan all but 1 ghost". that is fine.

Reply:
tests added, unused variable removed.